src/processing/processor.h

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright 2014-2024 by XGBoost Contributors
+ */
+#pragma once
+
+#include <any>
+#include <cstdint>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace processing {
+
+const char kLibraryPath[] = "LIBRARY_PATH";
+const char kMockProcessor[] = "mock";
+const char kLoadFunc[] = "LoadProcessor";
+
+/*! \brief An processor interface to handle tasks that require external library through plugins */
+class Processor {
+ public:
```

Comment:
Here is my understanding about these interfaces. Please confirm if it's correct:
ProcessGHPairs: active party encrypts g&h pairs
HandleGHPairs: passive parties save encrypted g&h pairs in local
ProcessAggregation: passive parties each cacluate it's histogram based on encrypted g&h pairs, the output is ciphertext
HandleAggregation: active party decrypts the allgathered historgram ciphertexts
ProcessHistograms: active party encrypted allgathered & decrypted histograms?
HandleHistograms: active parth decrypts the allgathered historgram ciphertexts?

I have two more questions:
1. Where are ProcessHistograms and HandleHistograms needed in secure xgboost?
2. Passive parties need a public key to perform HE addition calculations. The public key is generated by the active party. So I think there should be two more interfaces like ProcessPublicKey and HandlePublicKey.

Reply:
Hi @shaojian-ant , thanks for the question!
"ProcessHistograms" and "HandleHistograms" are for the upcoming secure horizontal scheme, I will remove these two from this PR to avoid confusion.
Under federated mode, the HE is supposed to be performed outside xgboost via the interface, hence the key management, etc. will not be part of xgboost lib (and hence in xgboost we only have a "mock" interface inplementation to illustrate the functionality)

Reply:
Thanks for your reply!!!